File: 0169-majority-element/0169-majority-element.ts
function majorityElement(nums: number[]): number {
    let returnValue:number = null;
    let count:number=0;

    for(let num of nums){
        if(count===0){
            returnValue=num;
        }
        if(returnValue===num){
            count++;
        }else{
            count--;
        }
    }
    return returnValue;
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int returnValue=0 ,count=0;
        for(int i=0;i<nums.length;i++){
            if( count==0){
                returnValue = nums[i];
            };
            if(returnValue==nums[i]){
                count++;
            }else{
                count--;
            }

        };
        return returnValue;
        
    }
}